File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
typedef vector<int> vi;
class Solution {
public:
    vector<int> rightSideView(TreeNode* r) {
        if(!r) return vi({});
        queue<TreeNode*> q;
        vi a;
        q.push(r);
        while(q.size() > 0){
            int sz = q.size();
            sz--;
            while(sz--){
                auto f= q.front();
                if(f->left) q.push(f->left);
                if(f->right)q.push(f->right);
                q.pop();
            }
            auto f= q.front();
                if(f->left) q.push(f->left);
                if(f->right)q.push(f->right);
            a.push_back(f->val);
            q.pop();
        }
        return a;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        q = deque()
        ans = []
        if root is None:
            return ans
        q.append(root)
        while len(q)>0:
            sz = len(q)
            lst = 0
            for _ in range(0,sz):
                u = q.popleft()
                lst = u.val
                for v in [u.left,u.right]:
                    if v:
                        q.append(v)
            ans.append(lst)
        return ans        